backend/src/main/java/ai/giskard/web/rest/controllers/ProjectController.java

```diff
@@ -132,4 +140,19 @@ public ProjectDTO invite(@PathVariable("id") Long id, @PathVariable("userId") Lo
         return giskardMapper.projectToProjectDTO(project);
     }
 
+    @PostMapping(value = "/project/import/{timestampDirectory}/{projectKey}/prepare", consumes = "multipart/form-data")
+    public PrepareImportProjectDTO prepareImport(@RequestParam("file") MultipartFile zipFile, @PathVariable("timestampDirectory") @NotNull String timestampDirectory, @PathVariable("projectKey") String projectKey, @AuthenticationPrincipal final UserDetails userDetails) throws IOException {
+        Path pathToTimestampDirectory;
+        permissionEvaluator.canWrite();
+        pathToTimestampDirectory = projectService.unzip(timestampDirectory, zipFile);
+        return projectService.prepareImport(pathToTimestampDirectory, projectKey);
+    }
+
+    @PostMapping(value = "/project/import/{timestampDirectory}/{projectKey}")
```

Comment:
After having another look at this code, I find it cleaner if frontend didn't have to be responsible for sending neither `timestampDirectory` nor `projectKey`. What we should only get from a user is a zip archive, for the rest:
- temporary directory name should be generated on the backend and sent to the frontend since it's the backend's responsibility to create that directory
- project key is taken from the archive directly and sent back to the frontend to autofill the project key text input